homeassistant/components/sensor/waze_travel_time.py

```diff
@@ -0,0 +1,177 @@
+"""
+Support for Waze travel time sensor.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.waze_travel_time/
+"""
+import logging
+from datetime import timedelta
+
+import requests
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['WazeRouteCalculator==0.5']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_DESTINATION = 'destination'
+CONF_ORIGIN = 'origin'
+CONF_OUTPUTS = 'outputs'
+CONF_REGION = 'region'
+CONF_UPDATE_INTERVAL = 'update_interval'
+
+TIME_UNIT = 'min'  # type: str
+DISTANCE_UNIT = 'km'  # type: str
+
+DEFAULT_NAME = 'Waze Travel Time'
+
+OUTPUT_TYPES = {
+    'duration':
+    ['Duration', TIME_UNIT, 'mdi:timer', 1],
+    'distance':
+    ['Distance', DISTANCE_UNIT, 'mdi:car', 1],
+    'route':
+    ['Route', None, 'mdi:car', 0],
+}
+
+REGIONS = ['US', 'NA', 'EU', 'IL']
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_ORIGIN): cv.string,
+    vol.Required(CONF_DESTINATION): cv.string,
+    vol.Required(CONF_OUTPUTS):
+        vol.All(cv.ensure_list, [vol.In(OUTPUT_TYPES)]),
+    vol.Optional(CONF_REGION, default='US'):
+        vol.All(cv.ensure_list, [vol.In(REGIONS)]),
+    vol.Optional(CONF_UPDATE_INTERVAL, default=timedelta(seconds=300)):
+        vol.All(cv.time_period, cv.positive_timedelta),
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the sensor platform."""
+    name = config.get(CONF_NAME)
+    origin = config.get(CONF_ORIGIN)
+    destination = config.get(CONF_DESTINATION)
+    region = config.get(CONF_REGION)[0]
```

Comment:
Why allow a list of regions in the config if we're only interested in the first item?